src/phases/query/CollScanPredicateSelectivity.yml

```diff
@@ -0,0 +1,688 @@
+ActorTemplates:
+- TemplateName: FindQueryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: "undefinedFilter"}}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: *Document
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Heuristic CE defines the following selectivity order for predicates:
+# $eq < closed range (e.g. {$gt: 10, $lte: 20}) < open range (e.g. {$gt: 10}) < $ne
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: HighSelHeuristicGoodTwoRangeClauses
+      ActivePhase: 3
+      Filter:
+        # Here (0, 1e9) is the most selective predicate, both in reality and according to heuristic CE.
+        &HighSelHeuristicGoodTwoRangeClauses {
+          onePercentPositiveInt: {$gt: 0, $lt: *Billion}, # Selectivity = .01
+          onePercentNegativeInt: {$gt: 0}, # Selectivity = .99
+        }
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: HighSelHeuristicGoodTwoEqClauses
+      ActivePhase: 4
+      Filter:
+        &HighSelHeuristicGoodTwoEqClauses {
+          onePercentNull: null, # Selectivity = .01
+          onePercentStr: {$ne: "non-null"}, # Selectivity = .99
+        }
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: MediumSelHeuristicGoodTwoRangeClauses
+      ActivePhase: 5
+      Filter:
+        &MediumSelHeuristicGoodTwoRangeClauses {
+          tenPercentPositiveInt: {$gt: 0, $lt: *Billion}, # Selectivity = .1
+          tenPercentNegativeInt: {$gt: 0}, # Selectivity = .9
+        }
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: MediumSelHeuristicGoodTwoEqClauses
```

Comment:
I think some of these names might be misleading, since $ne for e.g. is not an equality predicate. Could say EqNe instead or similar.